test/test_torch.py

```diff
@@ -13786,17 +13786,46 @@ def test_div(self, device, dtype):
                              0.01)
             self.assertEqual(a1.div(a2), a1 / a2)
 
-    @onlyCPU
     @dtypes(*torch.testing.get_all_math_dtypes('cpu'))
-    def test_floordiv(self, device, dtype):
-        if dtype is torch.float16:
+    def test_floor_divide_tensor(self, device, dtype):
+        if dtype == torch.half and self.device_type == 'cpu':
```

Comment:
wait, you still get torch.half here even though you did `get_all_math_dtypes('cpu')`?

Reply:
Good catch. You're correct this check should be unnecessary. I just copied from the existing test. I'll remove it.

Reply:
I separated the dtypes for CPU and CUDA so we're testing fp16 on CUDA and not on CPU, as desired.